darts/models/theta.py

```diff
@@ -115,14 +125,258 @@ def predict(self, n: int) -> 'TimeSeries':
+
+        # Store and remove seasonality effect if there is any.
+        if self.is_seasonal:
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period,
+                                                                model=self.season_mode)
+            new_ts = remove_from_series(new_ts, self.seasonality, model=self.season_mode)
+
+        ts_values = new_ts.univariate_values()
+        if (ts_values <= 0).any():
+            self.model_mode = ModelMode.ADDITIVE
+            self.trend_mode = TrendMode.LINEAR
+            logger.warning("Time series has negative values. Fallback to additive and linear model")
+
+        # Drift part of the decomposition
+        if self.trend_mode is TrendMode.LINEAR:
+            linreg = ts_values
+        else:
+            linreg = np.log(ts_values)
+        self.drift = np.poly1d(np.polyfit(np.arange(self.length), linreg, 1))
+        theta0_in = self.drift(np.arange(self.length))
+        if self.trend_mode is TrendMode.EXPONENTIAL:
+            theta0_in = np.exp(theta0_in)
+
+        if (theta0_in > 0).all() and self.model_mode is ModelMode.MULTIPLICATIVE:
+            theta_t = (ts_values ** self.theta) * (theta0_in ** (1 - self.theta))
+        else:
+            if self.model_mode is ModelMode.MULTIPLICATIVE:
+                logger.warning("Negative Theta line. Fallback to additive model")
+                self.model_mode = ModelMode.ADDITIVE
+            theta_t = self.theta * ts_values + (1 - self.theta) * theta0_in
+
+        # SES part of the decomposition.
+        self.model = hw.SimpleExpSmoothing(theta_t).fit()
+        theta2_in = self.model.fittedvalues
+
+        if (theta2_in > 0).all() and self.model_mode is ModelMode.MULTIPLICATIVE:
+            self.fitted_values = theta2_in**self.wses * theta0_in**self.wdrift
+        else:
+            if self.model_mode is ModelMode.MULTIPLICATIVE:
+                self.model_mode = ModelMode.ADDITIVE
+                logger.warning("Negative Theta line. Fallback to additive model")
+                theta_t = self.theta * ts_values + (1 - self.theta) * theta0_in
+                self.model = hw.SimpleExpSmoothing(theta_t).fit()
+                theta2_in = self.model.fittedvalues
+            self.fitted_values = self.wses * theta2_in + self.wdrift * theta0_in
+        if self.is_seasonal:
+            if self.season_mode is SeasonalityMode.ADDITIVE:
+                self.fitted_values += self.seasonality.univariate_values()
+            elif self.season_mode is SeasonalityMode.MULTIPLICATIVE:
+                self.fitted_values *= self.seasonality.univariate_values()
+        # Fitted values are the results of the fit of the model on the train series. A good fit of the model
+        # will lead to fitted_values similar to ts. But one cannot see if it overfits.
+        if self.normalization:
+            self.fitted_values *= self.mean
+        # Takes too much time to create a TimeSeries
```

Comment:
I think you can now remove this comment (or move to the corresponding backtesting function)